samza-core/src/main/java/org/apache/samza/zk/ZkLeaderElector.java

```diff
@@ -168,7 +163,7 @@ public void handleDataChange(String dataPath, Object data) throws Exception {
     @Override
     public void handleDataDeleted(String dataPath) throws Exception {
       LOGGER.info(zLog("Data deleted on path " + dataPath + ". Predecessor went away. So, trying to become leader again..."));
-      tryBecomeLeader();
+      tryBecomeLeader(leaderElectorListener);
```

Comment:
I thought we discussed this and you were going to fix this in one of your patches. But looks like we missed it. This is getting invoked in the event thread and should be invoked in a different thread.